streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -62,24 +75,60 @@ public void addChild(ProcessorNode<?, ?> child) {
         children.add(child);
     }
 
+
     public void init(ProcessorContext context) {
+        long startNs = time.nanoseconds();
         try {
             processor.init(context);
         } catch (Exception e) {
             throw new StreamsException(String.format("failed to initialize processor %s", name), e);
         }
+        this.nodeMetrics = new NodeMetrics(context.metrics(), name,  "task." + context.taskId());
+        this.nodeMetrics.metrics.recordLatency(nodeMetrics.nodeCreationSensor, startNs, time.nanoseconds());
     }
 
     public void close() {
+        long startNs = time.nanoseconds();
         try {
             processor.close();
+            nodeMetrics.metrics.recordLatency(nodeMetrics.nodeDestructionSensor, startNs, time.nanoseconds());
+            nodeMetrics.removeAllSensors();
         } catch (Exception e) {
             throw new StreamsException(String.format("failed to close processor %s", name), e);
         }
     }
 
     public void process(final K key, final V value) {
+        long startNs = -1;
+        // expensive operation. Only perform if we are recording this metric
+        if (nodeMetrics.nodeProcessTimeSensor.maybeRecord()) {
```

Comment:
There are a couple of other places where we can potentially save `nanoSeconds` calls (left comments on those places where `SENSOR_DEBUG` sensors may require getting the startNs and endNs). So I feel this could be a common pattern to add.